src/nile/cli.py

```diff
@@ -161,47 +208,67 @@ async def deploy(
 @click.option("--overriding_path")
 @click.option("--nile_account", is_flag=True)
 @network_option
-@mainnet_token_option
+@query_option
 @watch_option
+@enable_stack_trace
 async def declare(
+    ctx,
     signer,
     contract_name,
     network,
     max_fee,
-    watch_mode,
     alias,
     overriding_path,
-    token,
+    query,
     nile_account,
+    watch_mode,
 ):
     """Declare a StarkNet smart contract through an Account."""
-    account = await Account(signer, network, watch_mode=watch_mode)
-    await account.declare(
-        contract_name,
-        alias=alias,
-        max_fee=max_fee,
-        overriding_path=overriding_path,
-        mainnet_token=token,
-        watch_mode=watch_mode,
-        nile_account=nile_account,
-    )
+    account = await try_get_account(signer, network, watch_mode="track")
```

Comment:
Good catch! This is another of the temporary solutions to avoid separating Account initialization and deployment on this PR:

In setup, you are deploying the account itself, and in the rest, the account is being deployed if it is not already (because of initialization and deployment happening at the same time). That's why track is used in these cases, because the execution of the transaction won't work if the deploy_account transaction hasn't been accepted.

I think the account shouldn't be deployed by default on this cases, but an error should be given to the users instead. That's probably for the scope of the "separating initialization from deployment in account issue" though

Reply:
Agreed! I think returning an error feels like the way to go when the fetched account is None. And yeah, that's definitely for the scope of "separating initialization form deployment"